lib/iam_ssoe_oauth/session_manager.rb

```diff
@@ -15,8 +15,34 @@ def find_or_create_user
       create_user_session
     end
 
+    def logout
+      response = iam_ssoe_service.post_revoke(@access_token)
+      destroy_user_session if response.status == 200
+    end
+
     private
 
+    def destroy_user_session
+      uuid = @session.uuid
+
+      identity_destroy_count = IAMUserIdentity.find(uuid).destroy
+      user_destroy_count = IAMUser.find(uuid).destroy
+      session_destroy_count = @session.destroy
+
+      # redis returns number of records successfully deleted
+      if [identity_destroy_count, user_destroy_count, session_destroy_count].sum == 3
```

Comment:
Is it possible one of these counts gets returned as 2? That could throw off the sum and lead to a false positive.

Reply:
technically possible 😄 I had it as `.all? { |c| c == 1 } ` before I'll change it to != 0